pkg/datarepair/checker/checker.go

```diff
@@ -101,14 +104,31 @@ func (c *checker) identifyInjuredSegments(ctx context.Context) (err error) {
 					return Error.New("error getting offline nodes %s", err)
 				}
 				numHealthy := len(nodeIDs) - len(missingPieces)
-				if int32(numHealthy) < pointer.Remote.Redundancy.RepairThreshold {
+				if (int32(numHealthy) >= pointer.Remote.Redundancy.MinReq) && (int32(numHealthy) < pointer.Remote.Redundancy.RepairThreshold) {
 					err = c.repairQueue.Enqueue(&pb.InjuredSegment{
 						Path:       string(item.Key),
 						LostPieces: missingPieces,
 					})
 					if err != nil {
 						return Error.New("error adding injured segment to queue %s", err)
 					}
+				} else if int32(numHealthy) < pointer.Remote.Redundancy.MinReq {
+					// make an entry in to the irreparable table
+					rmtSegInfo := &irreparable.RemoteSegmentInfo{
+						SegmentPath:        item.Key,
+						SegmentDetail:      item.Value,
+						LostPiecesCount:    int64(len(missingPieces)),
+						RepairUnixSec:      time.Now().Unix(),
+						RepairAttemptCount: int64(1),
+					}
+
+					//check if that entry already exists
+					dbrmtSegInfo, err := c.irrdb.Get(ctx, rmtSegInfo)
```

Comment:
Why is this checking whether the entry exists? This increment should be atomic on the database side.

Reply:
remove and added the IncrementRepairAttempts()